### ISSUE TITLE
Can't add a block of data that references another .assets.resS as it gets erased.

### ISSUE THREAD

**Issue by zcxczxcz** [nesrak1/UABEA#393]
**Describe the bug**
While trying to add this block of data, whether through a Dump or by editing through UABEA:
 0 StreamingInfo m_StreamData
  0 UInt64 offset = 0
  0 unsigned int size = 49008
  1 string path = "mod.assets.resS"
It gets erased when importing the Dump of an object containing this block of data, or when forcing this block of data through editing an object and hitting "Ok" on the window in which you edit.

**To Reproduce**
1. Go to an asset the data of which does not contain a StreamingInfo block (in my case - a Mesh)
2. Click on "Edit Data"
3. Scroll down to the bottom, where this block wasn't imported
4. Paste the data block from the beginning of this bug report
5. Hit "Ok"
6. Enter editing mode again, the data block will be erased.

**Expected behavior**
The data block to be kept in so that the game reads the external .assets.resS file.

**Screenshots**
Dump file:
![image](https://github.com/user-attachments/assets/aaf2d50a-56f9-4b37-951e-b4dd6da51707)

Data in the UABEA Editor, deletes said data block every time i try to get it in:
![image](https://github.com/user-attachments/assets/6c332635-8faf-4235-ba75-3dbe7a316a31)

**Additional context**
While following a tutorial on replacing models in unity games. Trying to do it with the Raw Data method made the game crash. The StreamingInfo block is essential to load the external .assets.resS.

**Comment by nesrak1** [comment]
> Go to an asset the data of which does not contain a StreamingInfo block (in my case - a Mesh)

All assets in a bundle have to use the same structure for a given type, so if you have a mesh without streaming info then it just has to stay that way for every other mesh in the bundle. This is a bundle format thing, not UABEA.

When you import a text dump, the dump is turned into a raw dump and the contents of the asset are set to that raw data. This means your streamdata isn't technically being deleted, it's just that the type tree says this field doesn't exist so it doesn't read the rest of the asset data.

Also, are you using the same exact editor version to build the mesh?

**Comment by zcxczxcz** [comment]
Yes, the game is built on Unity 5.0.0 and my mesh was built in Unity 5.0.0.
Is there a way around this? Since importing the mesh as Raw Data just makes the game crash

**Comment by nesrak1** [comment]
What is the exact unity version used both in game and in editor? That stream data field does not exist in unity 5, but even if it did, it would have mesh metrics before it. So your first screenshot doesn't make any sense.

There is no "way around it". You have to match the format of mesh assets that engine version expects.
